pkg/cloud/amazon/ecr.go

```diff
@@ -19,11 +19,14 @@ import (
 // GetAccountIDAndRegion returns the current account ID and region
 func GetAccountIDAndRegion(profile string, region string) (string, string, error) {
 	sess, err := NewAwsSession(profile, region)
-	// We nee to get the region from the connected cluster instead of the one configured for the calling user
-	// as it might not be found and it would then use the default (us-west-2)
-	_, region, err = GetCurrentlyConnectedRegionAndClusterName()
-	if err != nil {
-		return "", "", err
+	if (region == "") {
+		// TODO for the original author (dgozalo): Is this right? We are setting up a new cluster - so we do not have a current cluster!?
```

Comment:
You are right. I think this function has been tweaked twice but only in the context of already being connected to a cluster as it was needed for things like ECR where it made sense to use the cluster region. 

It's right to use the provided region when it's not empty and use the cluster one when it is. We may have to make it more clear in the future, but for now we need this to let people create clusters.

Reply:
It's also true that, if you have your region configured in your local AWS credentials / configuration it should be taken as the region to create the cluster.